src/core/jit/exec.rs

```diff
@@ -147,6 +149,15 @@ where
                 &data_path,
                 result.map_err(|e| Positioned::new(e, field.pos)),
             );
+        } else {
+            // if the present field doesn't have IR, still go through it's extensions to see
+            // if they've IR.
+            join_all(field.nested_iter().map(|child| {
+                let ctx = ctx.with_field(child);
```

Comment:
how do you explain why for children of fields with IR we need to set the value, but for children of fields without IR don't?

I think it's a lack of proper test, because there should be no difference in setting the value if the IR is present or not.

you need to add test to verify that logic with deeper level of nesting.

I've prepared one:

``````md
# Recursive Type

```graphql @config
schema @server @upstream(baseURL: "https://jsonplaceholder.typicode.com") {
  query: Query
  mutation: Mutation
}

type User {
  name: String
  id: Int!
  connections: [Connection] @http(path: "/connections/{{.value.id}}")
}

type Connection {
  type: String
  nested: NestedUser
}

type NestedUser {
  user: User
}

type Query {
  user: User @http(path: "/users/1")
}

type Mutation {
  createUser(user: User): User @http(path: "/user", method: "POST", body: "{{.args.user}}")
}
```

```yml @mock
- request:
    method: GET
    url: https://jsonplaceholder.typicode.com/users/1
  response:
    status: 200
    body:
      id: 1
      name: User1
- request:
    method: GET
    url: https://jsonplaceholder.typicode.com/connections/1
  response:
    status: 200
    body:
      - type: friend
        nested:
          user:
            id: 2
            name: User2

- request:
    method: GET
    url: https://jsonplaceholder.typicode.com/connections/2
  response:
    status: 200
    body:
      - type: friend
        nested:
          user:
            id: 3
            name: User3
      - type: coworker
        nested:
          user:
            id: 4
            name: User4
```

```yml @test
- method: POST
  url: http://localhost:8080/graphql
  body:
    query: |
      query {
        user {
          name
          id
          connections {
            type
            nested {
              user {
                name
                id
                connections {
                  nested {
                    user {
                      name
                      id
                    }
                  }
                }
              }
            }
          }
        }
      }

```
``````